sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx

```diff
@@ -0,0 +1,352 @@
+        //     This contains the neutrino's parentage information
+        if (      fluxlist[inu].fptype==13  || fluxlist[inu].fptype==-13  ) ptype = 0;//mu
+        else if ( fluxlist[inu].fptype==211 || fluxlist[inu].fptype==-211 ) ptype = 1;//pi
+        else if ( fluxlist[inu].fptype==130                               ) ptype = 2;//K0
+        else if ( fluxlist[inu].fptype==321 || fluxlist[inu].fptype==-321 ) ptype = 3;//K
+        else {
+          throw cet::exception(__FUNCTION__) << GetName()<<"::Unknown ptype "<<fluxlist[0].fptype<< std::endl;
+        }
+
+        // Discover the neutrino type
+        //     This contains the neutrino's flavor information
+        if (      fluxlist[inu].fntype== 12  ) ntype=0;//nue
+        else if ( fluxlist[inu].fntype==-12  ) ntype=1;//nuebar
+        else if ( fluxlist[inu].fntype== 14  ) ntype=2;//numu
+        else if ( fluxlist[inu].fntype==-14  ) ntype=3;//numubar
+        else {
+          throw cet::exception(__FUNCTION__) << GetName()<<"::Unknown ntype "<<fluxlist[inu].fntype<< std::endl;
+        }
+
+        // Collect neutrino energy
+        double enu= mclist[inu].GetNeutrino().Nu().E();
+
+        //Let's make a weights based on the calculator you have requested 
+
+        if(fParameterSet.fRWType == EventWeightParameterSet::kMultisim){
+
+          for (size_t i=0;i<weights.size();i++) {
+            double randomN = (fParameterSet.fParameterMap.begin())->second[i];
+            weights[i]=UnisimWeightCalc(enu, ptype, ntype, randomN , PosOnly);//AddParameter result does not work here;
+
+            if(weights[i]<0){
+              wcn++;
+            }else if((weights[i]-0)<1e-30){
+              wc0++;
+            }else if(fabs(weights[i]-30) < 1e-30){
+              wc30++;
+            } else if(fabs(weights[i]-1)<1e-30){
+              wc1++;
+            } else {
+              wc++;
+            }
+          }//Iterate through the number of universes      
+        }
+      } else{//then this must be PrimaryHadron
+
+
+        // First let's check that the parent of the neutrino we are looking for is 
+        //  the particle we intended it to be, if not set all weights to 1
+      if (std::find(fprimaryHad.begin(), fprimaryHad.end(),(fluxlist[inu].ftptype)) == fprimaryHad.end() ){//if it does not contain any particles we need get 1
+          weights.resize( NUni);
+          std::fill(weights.begin(), weights.end(), 1);
+          return weights;//done, all 1
+        }// Hadronic parent check
+
+        if(fParameterSet.fRWType == EventWeightParameterSet::kMultisim){
+
+          for (unsigned int i = 0; int(weights.size()) < NUni; i++) {//if all weights are 1, no need to calculate weights;
+            std::pair<bool, double> test_weight;
+
+            std::vector< float > Vrandom = (fParameterSet.fParameterMap.begin())->second;//vector of random #
+            std::vector< float > subVrandom;//sub-vector of random numbers;
```

Comment:
why make these `std::vector <float>` instead of `std::vector<double>`

Reply:
Same as above, use float to reduce the output root size.